sycl/doc/extensions/experimental/sycl_ext_oneapi_kernel_compiler.asciidoc

```diff
@@ -172,22 +178,34 @@ a!
 ----
 namespace sycl::ext::oneapi::experimental {
 
+template<typename PropertyListT = empty_properties_t>
 kernel_bundle<bundle_state::ext_oneapi_source> create_kernel_bundle_from_source(
   const context& ctxt,
   source_language lang,
-  const std::string& source)
+  const std::string& source,
+  PropertyListT props = {})
 
 } // namespace sycl::ext::oneapi::experimental
 ----
 !====
 
+_Constraints:_ Available only when `PropertyListT` is an instance of
+`sycl::ext::oneapi::experimental::properties` which contains no properties
+other than those listed below in the section "New properties for the
+`create_kernel_bundle_from_source` function".
+
 _Effects:_ Creates a new kernel bundle that represents a kernel that is defined
 by the source code string `source` using the language `lang`.
 The bundle is associated with the context `ctxt`, and kernels from this bundle
 may only be submitted to a queue that shares the same context.
 The bundle's set of associated devices is the set of devices contained in
 `ctxt`.
 
+When the source language is `source_language::sycl`, the `source` string must
+not `#include <sycl/sycl.hpp>`.
```

Comment:
That's a good point.  We could relax this to say that the `source` string does not need to `#include <sycl/sycl.hpp>`, while still allowing it.

Reply:
I think this would be better. But even saying that it isn't needed might have a disadvantage in the future. In the future we might want to split sycl.hpp into different parts to improve compilation speed. And then online compile wouldn't benefit from it because we would need to include all of the device side API to be backward compatible. I'm not sure the convenience of not having to include sycl.hpp is worth this potential downside in the future. It might be best only to say that sycl.hpp is automatically included in `include_files` so that the user doesn't have to do that part.

Reply:
I think the user's source string should explicitly include `#include <sycl/sycl.hpp>`.  I agree with @rolandschulz that we may want to change the headers in future, and it would be nice to have this flexibility.

I also think that it's weird to make an exception for the SYCL headers, because users may need to include other C++ headers (e.g., `#include <cmath>`) if they're making use of any `std::` functionality inside of their kernels.

I think we should replace this with a note to the effect that the compiler behaves as-if the location of the SYCL headers and standard C++ headers are included in `include_files`.  Do we also need to add something saying that these headers are guaranteed to be the same as those used when the host program was compiled? (Can we actually guarantee that?)

Reply:
OK, I agree that this makes sense.  Changed in d49df130d19fda8f42a8d0735c5572d434c01455.